File: src/main/java/me/m64diamondstar/ingeniamccore/Wands/WandPlayer.java
package me.m64diamondstar.ingeniamccore.Wands;

import org.bukkit.entity.Player;

public class WandPlayer {

    private Player player;
    private long cooldown;
    private boolean flying;

    public WandPlayer(Player player){
        this.player = player;
    }

    public void setCooldown(Long op, Long vipp, Long vip, Long visitor){
        if(player.hasPermission("ingeniawands.owner") || player.isOp()) {
            cooldown = System.currentTimeMillis() + (op);
        } else if(player.hasPermission("ingenia.vip+")) {
            cooldown = System.currentTimeMillis() + (vipp);
        } else if(player.hasPermission("ingenia.vip")) {
            cooldown = System.currentTimeMillis() + (vip);
        } else {
            cooldown = System.currentTimeMillis() + (visitor);
        }
    }

    public long getCooldown(){
        return cooldown;
    }

    public void setFlying(){
        this.flying = true;
    }

    public void setFlying(boolean flying){
        this.flying = flying;
    }



}

File: src/main/java/me/m64diamondstar/ingeniamccore/wands/WandPlayer.kt
package me.m64diamondstar.ingeniamccore.wands

import java.util.HashMap
import me.m64diamondstar.ingeniamccore.wands.Cooldowns
import org.bukkit.entity.Player

class WandPlayer(private val player: Player) {
    var cooldown: Long = 0
        private set
    private var flying = false
    fun setCooldown(op: Long, vipp: Long, vip: Long, visitor: Long) {
        cooldown = if (player.hasPermission("ingeniawands.owner") || player.isOp) {
            System.currentTimeMillis() + op
        } else if (player.hasPermission("ingenia.vip+")) {
            System.currentTimeMillis() + vipp
        } else if (player.hasPermission("ingenia.vip")) {
            System.currentTimeMillis() + vip
        } else {
            System.currentTimeMillis() + visitor
        }
    }

    fun setFlying() {
        flying = true
    }

    fun setFlying(flying: Boolean) {
        this.flying = flying
    }
}